File: code_place/PythonCode/src/two_num.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        tmp_dir = {}

        for i in range(len(nums)):
            tmp_target = target - nums[i]

            if tmp_target in tmp_dir.keys():
                return [tmp_dir.get(tmp_target), i]

            tmp_dir[nums[i]] = i

        return []


if __name__ == '__main__':
    s = Solution()
    print(s.twoSum([2, 7, 11, 15], 9))
    print(s.twoSum([3, 2, 4], 6))
    print(s.twoSum([3, 3], 6))

File: code_place/GoCode/two_num.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	tmpMap := make(map[int]int, 0)
	for i, item := range nums {
		tmpTarget := target - item

		if index, ok := tmpMap[tmpTarget]; ok {
			return []int{index, i}
		}

		tmpMap[item] = i
	}

	return nil
}

func main() {

	arr1 := []int{2, 7, 11, 15}
	arr2 := []int{3, 2, 4}
	arr3 := []int{3, 3}
	fmt.Printf("%v\n", twoSum(arr1, 9))
	fmt.Printf("%v\n", twoSum(arr2, 6))
	fmt.Printf("%v\n", twoSum(arr3, 6))
}
